File: sdk/python/example_app.py
from codex_sdk import CodexClient
import pathlib

db_path = pathlib.Path(__file__).parent / 'example-db.db'
client = CodexClient(str(db_path))

# Set a value
client.set('greeting', {'msg': 'Hello from Python!'})

# Get the value
value = client.get('greeting')
print('Greeting:', value['msg'])

# List all keys
print('All keys:', client.keys())

# Delete the key
deleted = client.delete('greeting')
print('Deleted:', deleted)

# Cleanup
db_path.unlink(missing_ok=True)

File: sdk/nodejs/example_app.js
const path = require('path');
const CodexClient = require('./index');

const dbPath = path.resolve(__dirname, 'example-db.db');
const client = new CodexClient({ file: dbPath });

(async () => {
  // Set a value
  await client.set('greeting', { msg: 'Hello from Node.js!' });

  // Get the value
  const value = await client.get('greeting');
  console.log('Greeting:', value.msg);

  // List all keys
  const keys = await client.keys();
  console.log('All keys:', keys);

  // Delete the key
  const deleted = await client.delete('greeting');
  console.log('Deleted:', deleted);

  // Cleanup
  const fs = require('fs');
  try { fs.unlinkSync(dbPath); } catch (e) {}
})();
